cocos/ui/UIScale9Sprite.cpp

```diff
@@ -105,15 +113,20 @@ namespace ui {
     bool Scale9Sprite::initWithSpriteFrameName(const std::string& spriteFrameName,
                                                const Rect& capInsets)
     {
-        CCASSERT((SpriteFrameCache::getInstance()) != NULL,
-                 "SpriteFrameCache::getInstance() must be non-NULL");
-
-        SpriteFrame *frame = SpriteFrameCache::getInstance()->getSpriteFrameByName(spriteFrameName);
-        CCASSERT(frame != NULL, "CCSpriteFrame must be non-NULL");
-
-        if (NULL == frame) return false;
-        bool pReturn = this->initWithSpriteFrame(frame, capInsets);
-        return pReturn;
+        bool ret = false;
+        do {
+            auto spriteFrameCache = SpriteFrameCache::getInstance();
+            CCASSERT(spriteFrameCache != nullptr,
+                     "SpriteFrameCache::getInstance() must be non-NULL");
+            
+            SpriteFrame *frame = spriteFrameCache->getSpriteFrameByName(spriteFrameName);
+            CCASSERT(frame != nullptr, "CCSpriteFrame must be non-NULL");
+            if (frame == nullptr) return false;
```

Comment:
`  if (frame == nullptr) return false;` should be `  if (frame == nullptr) break`?